app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-core.js.erb

```diff
@@ -13,6 +13,15 @@ window.GOVUK.analyticsGA4 = window.GOVUK.analyticsGA4 || {};
       'https://www.googletagmanager.com/gtm.js?id='+i+dl+ '&gtm_auth=' + window.GOVUK.analyticsGA4.vars.auth + '&gtm_preview=' + window.GOVUK.analyticsGA4.vars.preview + '&gtm_cookies_win=x';f.parentNode.insertBefore(j,f);
       })(window,document,'script','dataLayer',window.GOVUK.analyticsGA4.vars.id);
       /* eslint-enable */
+    },
+
+    sendData: function (data) {
+      data.govuk_gem_version = this.getGemVersion()
+      window.dataLayer.push(data)
```

Comment:
Quick question; would we potentially want to wrap this within a guard to check whether the `dataLayer` exists?

Reply:
I thought about that but for code using this it's a bit late in the process - most of our code checks for the existence of this already at a much earlier point, e.g. before setting up event listeners and other variables.